File: rb/features.rb
# ShameAsAService SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module ShameAsAServiceFeatures
  def self.make_feature(name)
    case name
    when "base"
      ShameAsAServiceBaseFeature.new
    when "ratelimit"
      ShameAsAServiceRatelimitFeature.new
    when "retry"
      ShameAsAServiceRetryFeature.new
    when "test"
      ShameAsAServiceTestFeature.new
    when "timeout"
      ShameAsAServiceTimeoutFeature.new
    else
      ShameAsAServiceBaseFeature.new
    end
  end
end

File: py/features.py
# ShameAsAService SDK feature factory

from feature.base_feature import ShameAsAServiceBaseFeature
from feature.test_feature import ShameAsAServiceTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ShameAsAServiceBaseFeature(),
        "test": lambda: ShameAsAServiceTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
